### PR TITLE
feat: Implement comprehensive choice selection system (#69)

### PR DESCRIPTION
**Pull request by KirkDiggler**
## Summary
Implements Issue #69: Extend choice selection system beyond skills to support fighting styles, cantrips, spells, and other class-specific choices for D&D character creation.

## Features Added
- **New Choice System Entities**: Complete type system for choices, categories, options, and validation
- **Service Interface Extensions**: Added `UpdateChoices` and `ListChoiceOptions` methods with proper Input/Output types
- **Progress Tracking**: Extended creation flow with `ProgressStepChoices` (bitflag 128)
- **Class-Specific Choices**: 
  - Fighter: Fighting style selection (6 options: Archery, Defense, Dueling, etc.)
  - Wizard: 3 cantrips + 6 level 1 spells from spell list
  - Cleric: 3 cantrips from spell list  
  - Sorcerer: 4 cantrips + 2 level 1 spells from spell list
- **External API Integration**: Dynamic spell options via D&D 5e API client
- **Security-First Validation**: Integer overflow protection and comprehensive validation

## Technical Implementation
- **DRY Principle**: Eliminated code duplication with `createSpellChoiceCategory` helper
- **Type Safety**: All string constants properly defined, Input/Output types for all methods
- **Entity Design**: Clean separation between choice entities and character draft
- **Validation Engine**: Prerequisites, conflicts, bounds checking with specific error codes
- **Mock Generation**: Updated gomock interfaces for testing

## Security Considerations
- **Overflow Protection**: Validates option count before int32 conversion to prevent attacks
- **Bounds Checking**: Validates min/max choices against malicious inputs
- **Error Handling**: Specific error codes for different validation failures

## Test Plan
- [x] All existing tests pass (73.5% coverage maintained)
- [x] Linter clean (0 issues)
- [x] Pre-commit workflow passes
- [x] Service interface methods properly mocked
- [x] Choice validation with various edge cases

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: internal/orchestrators/character/orchestrator.go [hunk 1]
```diff
@@ -1516,6 +1521,608 @@ func convertValidationErrorsToWarnings(errors []engine.ValidationError) []Valida
...
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get class details")
+	}
+
+	class := classDetails.Class
+
+	// Add class-specific choices based on class ID
+	switch draft.ClassID {
+	case "fighter":
+		if filterType == nil || *filterType == dnd5e.ChoiceTypeFightingStyle {
```
Comment (KirkDiggler) [inline internal/orchestrators/character/orchestrator.go:1817]: can we do this filtertype nil checks differently? do we have to dereference it? can we do that one time?
